lib/provider/home_screen_provider.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:e_commerce_application/repository/product_repository.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:e_commerce_application/models/product.dart';
+import 'package:e_commerce_application/services/services_configuration.dart';
+
+class HomeScreenProvider extends ChangeNotifier {
+  final ProductRepository productRepository = getIt<ProductRepository>();
+  List<Product> productList = [];
+  String _selectedOption = 'Regular';
+
+  HomeScreenProvider() {
+    fetchData();
```

Comment:
Refrain from fetching data when initialising the provider. Call fetchData function from Widget when needed.